### PR TITLE
Index data getter with out params

### PR DESCRIPTION
**Pull request by signorpipo**

### CODE REVIEW

File: src/wonderland.ts [hunk 1]
```diff
@@ -2490,6 +2490,51 @@ export class Mesh {
         return this._engine.wasm._wl_mesh_get_vertexCount(this._index);
     }
 
+    /** Number of indices in this mesh, or 0 if the mesh is not indexed.  */
+    get indexCount(): number {
```
Comment (DavidPeicho) [inline src/wonderland.ts:2494]: Better to return `null` if it's not indexed

File: src/wonderland.ts [hunk 2]
```diff
@@ -2510,6 +2555,55 @@ export class Mesh {
...
+    getIndexData(out?: Uint8Array | Uint16Array | Uint32Array | null): Uint8Array | Uint16Array | Uint32Array | null {
+        const wasm = this._engine.wasm;
+        const tempMem = wasm._tempMem;
+        const ptr = wasm._wl_mesh_get_indexData(this._index, tempMem, tempMem + 4);
+        if (ptr === null) return null;
+
+        const indexCount = wasm.HEAPU32[tempMem / 4];
+        const indexSize = wasm.HEAPU32[tempMem / 4 + 1];
+
+        if (!out) out = Mesh.createMeshIndexArray(indexSize, indexCount);
```
Comment (DavidPeicho) [inline src/wonderland.ts:2577]: I feel like we should throw if `out` has a smaller bit depth than what's expected.

Comment (SignorPipo) [reply src/wonderland.ts:2577]: How would you check for that?

Comment (DavidPeicho) [reply src/wonderland.ts:3100]: If it's an array buffer, you could check for `out.buffer.BYTES_PER_ELEMENT` or something like that

File: src/wonderland.ts [hunk 3]
```diff
@@ -3004,7 +3004,59 @@ export class Mesh {
...
+        const wasm = this._engine.wasm;
+        const tempMem = wasm._tempMem;
+        const ptr = wasm._wl_mesh_get_indexData(this._index, tempMem, tempMem + 4);
+        if (ptr === null) return null;
+
+        return wasm.HEAPU32[tempMem / 4];
+    }
+
+    /** The mesh index number type, or `null` if the mesh is not indexed.  */
+    get meshIndexType(): number | null {
```
Comment (DavidPeicho) [inline src/wonderland.ts:3018]: I think `indexType` is fine, doesn't need to be called `meshIndexType`.

Could you please return a `MeshIndexType` value:

```ts
get indexCount(): MeshIndexType | null {
    ...
    return wasm.HEAPU32[tempMem / 4] as MeshIndexType;
}
```

**Comment by DavidPeicho** [comment]
Allocating a TypedArray is probably around `64` bytes, maybe a bit more depending on the implementation. However, copying index data is probably moderately large in a lot of use cases. Thus, here I would recommend:
* Keep the indexCount and meshIndexType getter, no harm here
* Remove the `getIndexData`
* Add a new getter called `indexDataView,` keep the code from `get indexData` in this
* Make `get IndexData` use `get indexDataView`, and add a deprecation message stating to use get indexDataView instead.

**Pull request closed without merging**